sdks/python/apache_beam/ml/inference/model.py

```diff
@@ -0,0 +1,74 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import abc
+
+VALID_PYTORCH_DEVICE_TYPES = ['CPU', 'GPU']
```

Comment:
Should these be class-level variables on the pytorch/sklearn implementations?

Reply:
Yeah, probably. The constants should never be used outside this module (no need to import), so it probably makes more sense to put it in the class. 

Curious, is this more of a stylistic choice?
According to [PEP8](https://www.python.org/dev/peps/pep-0008/#constants), constants are _usually_ defined on a module level, but doesn't say that it needs to be class level. 

I suppose some downsides to class-level variables is the possibility of them be shadowed by subclasses.

Reply:
This is a stylistic choice, a question of trying to keep things local. One way to think about this is taking things to the limit, e.g. if one were to add hundreds of model types, would it make sense to enumerate all their constants here (and the classes below) or keep the constants local to the class. There's also the question of namespacing--the fact that you have one constant named _PYTORCH_ and another named _SKLEARN_ make it feel like there should be namespaces here. (Not hard and fast rules of course, but food for thought.) 

The possibility of shadowing by a subclass is actually a feature--if self.VALUE is used a subclass could decide to extend this list, if cls.VALUE is used one always gets the unshadowed version. 

All of this is probably moot if they are moved to their own modules.

Reply:
Makes sense! and yeah I'll end up refactoring to their own modules :)